### PR TITLE
Improve auth status text and JSON output

### LINKED ISSUE #397: Improve auth status text and JSON output
Objective:
- Improve `grace auth status` so human output starts with a localized authentication summary and structured output gives agents a useful status object instead of only the active-source string.

Context and evidence:
- Current human output starts with `GRACE_TOKEN: False` and ends with `Active source: Interactive (cached token)`.
- Current JSON output for `grace auth status -o Json` returns only `ReturnValue: "Interactive (cached token)"`, which hides whether the CLI is authenticated, which sources are configured, and what warnings/errors matter.
- Implementation surface found in `src/Grace.CLI/Command/Auth.CLI.fs` around the `Status` command.
- CLI tests should live in `src/Grace.CLI.Tests`, preferably the auth-focused test surface.

Owned paths:
- `src/Grace.CLI/Command/Auth.CLI.fs`
- `src/Grace.CLI.Tests/Auth*.fs`
- `src/Grace.CLI.Tests/Grace.CLI.Tests.fsproj` only if a new test file is needed
- Nearby CLI docs only if runtime behavior docs already describe `auth status` output

Forbidden or sensitive paths:
- Server auth, PAT validation, OIDC flow, secure-store implementation, and token storage behavior are out of scope unless tests prove the output change cannot be isolated.
- Do not print or serialize access tokens, refresh tokens, client secrets, raw PATs, or credential-bearing values.
- Do not change auth precedence: valid `GRACE_TOKEN`, invalid `GRACE_TOKEN`, M2M, interactive, none.

Risk surfaces:
- Auth or secrets
- CLI public contract
- Docs or workflow

Minimum detail gate:
- Invariant tuple: `grace auth status` reports the current CLI authentication state derived from environment PAT shape, OIDC M2M config, interactive CLI config, secure-store availability, and cached interactive token presence; `activeSource` remains the source selected by the existing precedence; JSON `ReturnValue` is the durable automation contract for the same source-of-truth values.
- Forbidden implementation shapes: do not infer authentication from configuration alone; do not mark invalid `GRACE_TOKEN` as authenticated; do not expose token contents or secret-bearing config; do not special-case only English text in a way that bypasses existing localization/resource patterns; do not break `--output Json`, `--select`, `--schema`, or stdout cleanliness; do not move auth precedence or side effects out of the status command without tests.
- Expected tests:
  - positive: interactive cached token or valid PAT produces `Authenticated`, a blank line, then bold/color `Active source` before the existing detail lines in human output; JSON `ReturnValue` contains `Authenticated = true`, `Status`, `ActiveSource`, source booleans, expiry/subject when present, and diagnostics/warnings/errors when applicable.
  - negative: no valid credential/source produces `Not authenticated` and structured JSON with `Authenticated = false` and an explicit active source of `None`.
  - regression: invalid `GRACE_TOKEN` remains visible as the active source/error but does not mark the status as authenticated, and no raw token/secret appears in text or JSON.
  - boundary: interactive configured with no cached token, secure-store unavailable, and config errors keep distinct machine-readable fields without collapsing into one string.
- High-risk adversarial examples: invalid PAT-shaped environment values, blank env vars, expired cached interactive token if existing status logic exposes expiry without refreshing, secure-store exceptions, `--select ReturnValue.ActiveSource`, JSON schema/example drift, and localized text labels.
- Selected risk-surface traps: CLI mode/side-effect interaction for `--output Json`, `--select`, `--schema`; auth redaction; false-positive tests that only assert substrings but miss output order; contract drift in command-output metadata if this command has a declared contract.
- Explicit N/A waivers with reasons: server/API authorization is N/A because this is CLI local status rendering only; Aspire/full validation is N/A unless implementation unexpectedly touches server/runtime integration; OpenAPI/SDK are N/A because no HTTP or SDK contract changes are intended.

Validation:
- Focused command: build `src/Grace.CLI.Tests/Grace.CLI.Tests.fsproj` in Release, then run targeted auth-status tests with `dotnet test --no-build --configuration Release src/Grace.CLI.Tests/Grace.CLI.Tests.fsproj --filter <auth status filter>`.
- Fast repo gate: `pwsh ./scripts/validate.ps1 -Fast`.
- Full or Aspire gate, if needed: N/A unless touched paths expand into Aspire/runtime/server surfaces.
- Manual verification, if needed: `grace auth status` and `grace auth status -o Json` from a built CLI if practical, recording that no token values are printed.

Definition of done:
- Human `auth status` begins with localized `Authenticated` or `Not authenticated`, then a blank line, then bold/color `Active source`, then the remaining details.
- JSON `ReturnValue` is a structured object useful for agents and includes auth status, active source, source/config/token booleans, expiry/subject when available, and diagnostics without secrets.
- Tests cover positive, negative, regression, and boundary cases or explicitly justify any untestable case.
- Coding and fix work completed through implementation subagents, with the main agent acting as orchestrator.
- Worker completed a bot-prevention self-review over the actual diff before handoff.
- Ready-for-review pull request opened and linked.
- Validation recorded.
- Codex Code Review Bot reviewed the latest PR commit and either reported no issues with a thumbs-up reaction on the PR body or all bot comments were fixed, answered, resolved, and followed by a no-issues bot review.
- Follow-ups named.

### PR DESCRIPTION
**Pull request by ScottArbeit**
Closes #397

## Summary

- Updates `grace auth status` human output so the first line is the authentication summary, followed by a blank line and the emphasized active source before the remaining details.
- Replaces the JSON `ReturnValue` string with a structured auth-status object containing authentication state, active source, source/config/token booleans, optional expiry/subject, and diagnostics without token or secret values.
- Adds focused CLI auth-status tests covering authenticated, unauthenticated, invalid `GRACE_TOKEN`, and JSON output behavior.

## Review Status

- Implementation worker: completed in `3cb89c2ba9d0d214702fcbef71e96ba29d8372eb`; branch rebased onto `9889bb3` and republished as `0c35cd5`; review fixes pushed in `b85881f2927f071928816dfb4c6b46abbe209c99`.
- Codex Code Review Bot: original three findings fixed in `b85881f`; follow-up JSON boolean and secure-store findings fixed in `59a414e`; Codex Code Review Bot reported no issues with a thumbs-up reaction on latest head `7ffdcd8`.
- Open bot findings: none currently unresolved after `59a414e`; maintainer follow-up commit `7ffdcd8` updates `scripts/validate.ps1`; GitHub `full` passed on latest head and Codex Code Review Bot reported no issues with a thumbs-up reaction. The `NU1605` package downgrade was fixed in `b85881f` by aligning `Grace.CLI.LocalStateDb.Worker` to `FSharp.Core` `10.1.301`.

## Validation

- `dotnet fantomas src\Grace.CLI\Command\Auth.CLI.fs src\Grace.CLI.Tests\Auth.Tests.fs`: passed.
- Pre-rebase `dotnet test src\\Grace.CLI.Tests\\Grace.CLI.Tests.fsproj --configuration Release -p:TreatWarningsAsErrors=false --filter AuthTests`: passed, 7 passed, 0 failed.\n- Post-rebase `dotnet test src\\Grace.CLI.Tests\\Grace.CLI.Tests.fsproj --configuration Release --filter AuthTests`: failed during restore on `NU1605` FSharp.Core downgrade before tests ran.
- `git diff --check`: passed.
- Pre-rebase `pwsh ./scripts/validate.ps1 -Fast`: blocked before tests ran during restore/build because `MessagePack` `3.1.6` was reported as `NU1903`.\n- Post-fix `pwsh ./scripts/validate.ps1 -Fast`: passed; build succeeded with 0 warnings and 0 errors; selected fast suites passed (Authorization 39, Types 96, Server.Unit 498, CLI 508).

## Docs Impact

No docs changed. Nearby auth docs describe command usage and authentication configuration, not the previous `auth status` output order or JSON `ReturnValue` shape.

## Residual Risk

- The package-update `main` commit removed the prior `MessagePack` `NU1903` blocker. The `FSharp.Core` `NU1605` downgrade in `Grace.CLI.LocalStateDb.Worker` was fixed in this PR because it blocked validation on the refreshed branch.
- `--schema` and routed `--select` metadata were not expanded in this slice; the runtime JSON `ReturnValue` is now structured, and broader command-output registry work can be tracked separately if Grace wants formal schema/select metadata for `auth status`.

### CODE REVIEW

File: src/Grace.CLI/Command/Auth.CLI.fs [hunk 1]
```diff
@@ -1196,12 +1245,75 @@ module Auth =
...
+                        | None -> ()
+
+                        match configError with
+                        | Some message -> yield { Level = "Error"; Source = "Auth config"; Message = message }
+                        | None -> ()
+                    ]
+
+                let authStatus =
+                    {
+                        Authenticated = authenticated
```
[the remark this answers is not in the document]
Comment (ScottArbeit) [reply src/Grace.CLI/Command/Auth.CLI.fs]: ## Review/Fix: Preserve false auth status booleans in JSON

**Status:** Fixed in 59a414e
**Review source:** Codex Code Review Bot
**Validation:** dotnet test --configuration Release src/Grace.CLI.Tests/Grace.CLI.Tests.fsproj --filter "FullyQualifiedName~Auth" passed 13/13; pwsh ./scripts/validate.ps1 -Fast passed; git diff --check passed.
**Prevention:** contract-propagation gap; current issue/PR review ledger updated, no template change needed.

### Summary

Fixed in `59a414e`. Auth status DTO booleans are now represented so explicit `false` values serialize under Grace shared JSON options, with regressions requiring false properties to be present for no-source and invalid-token cases.

### Review Issue

The bot identified a JSON diagnostics contract issue in src/Grace.CLI/Command/Auth.CLI.fs.

### Fix

Implemented by the review-fix worker in commit 59a414e0826a47be7b3bae28770ace1e1351e225, with focused regressions in src/Grace.CLI.Tests/Auth.Tests.fs.

### Validation

- dotnet fantomas src/Grace.CLI/Command/Auth.CLI.fs src/Grace.CLI.Tests/Auth.Tests.fs: passed.
- dotnet test --configuration Release src/Grace.CLI.Tests/Grace.CLI.Tests.fsproj --filter "FullyQualifiedName~Auth": passed, 13/13.
- pwsh ./scripts/validate.ps1 -Fast: passed.
- git diff --check: passed.

File: src/Grace.CLI/Command/Auth.CLI.fs [hunk 2]
```diff
@@ -1173,6 +1219,7 @@ module Auth =
 
                 let interactiveConfigured = cliConfig |> Option.isSome
                 let interactiveTokenPresent = interactiveBundle |> Option.isSome
+                let secureStoreAvailable = secureStoreError.IsNone
```
[the remark this answers is not in the document]
Comment (ScottArbeit) [reply src/Grace.CLI/Command/Auth.CLI.fs]: ## Review/Fix: Report secure storage only after checking it

**Status:** Fixed in 59a414e
**Review source:** Codex Code Review Bot
**Validation:** dotnet test --configuration Release src/Grace.CLI.Tests/Grace.CLI.Tests.fsproj --filter "FullyQualifiedName~Auth" passed 13/13; pwsh ./scripts/validate.ps1 -Fast passed; git diff --check passed.
**Prevention:** contract-propagation gap; current issue/PR review ledger updated, no template change needed.

### Summary

Fixed in `59a414e`. `SecureStoreAvailable` now requires interactive auth to be configured and secure-store verification to succeed, so unauthenticated/no-config diagnostics no longer claim secure storage is available.

### Review Issue

The bot identified a JSON diagnostics contract issue in src/Grace.CLI/Command/Auth.CLI.fs.

### Fix

Implemented by the review-fix worker in commit 59a414e0826a47be7b3bae28770ace1e1351e225, with focused regressions in src/Grace.CLI.Tests/Auth.Tests.fs.

### Validation

- dotnet fantomas src/Grace.CLI/Command/Auth.CLI.fs src/Grace.CLI.Tests/Auth.Tests.fs: passed.
- dotnet test --configuration Release src/Grace.CLI.Tests/Grace.CLI.Tests.fsproj --filter "FullyQualifiedName~Auth": passed, 13/13.
- pwsh ./scripts/validate.ps1 -Fast: passed.
- git diff --check: passed.
